### PR TITLE
[SPARK-35884][SQL] EXPLAIN FORMATTED for AQE

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

This is a followup of https://github.com/apache/spark/pull/29137 , which has some issues when running EXPLAIN FORMATTED
```
AdaptiveSparkPlan (13)
+- == Final Plan ==
   * HashAggregate (12)
   +- CustomShuffleReader (11)
      +- ShuffleQueryStage (10)
         +- Exchange (9)
            +- * HashAggregate (8)
               +- * Project (7)
                  +- * BroadcastHashJoin Inner BuildRight (6)
                     :- * LocalTableScan (1)
                     +- BroadcastQueryStage (5)
                        +- BroadcastExchange (4)
                           +- * Project (3)
                              +- * LocalTableScan (2)
+- == Initial Plan ==
   HashAggregate (unknown)
   +- Exchange (unknown)
      +- HashAggregate (unknown)
         +- Project (unknown)
            +- BroadcastHashJoin Inner BuildRight (unknown)
               :- Project (unknown)
               :  +- LocalTableScan (unknown)
               +- BroadcastExchange (unknown)
                  +- Project (3)
                     +- LocalTableScan (2)
```

Some nodes do not have an ID and show `unknown`. This PR fixes the issue.

### Why are the changes needed?

bug fix

### Does this PR introduce _any_ user-facing change?

EXPLAIN FORMATTED with AQE displays correctly.

### How was this patch tested?

new tests

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @allisonwang-db @peter-toth @viirya @maropu

File: sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala [hunk 1]
```diff
@@ -547,30 +547,107 @@ class ExplainSuiteAE extends ExplainSuiteHelper with EnableAdaptiveExecutionSuit
...
+      """
+        |(21) AdaptiveSparkPlan
+        |Output [4]: [k#x, count(v1)#xL, sum(v1)#xL, avg(v2)#x]
+        |Arguments: isFinalPlan=true
+        |""".stripMargin
     )
+    checkKeywordsNotExistsInExplain(testDf, FormattedMode, "unknown")
+  }
+
+  test("SPARK-35884: Explain should only display one plan before AQE takes effect") {
```
Comment (viirya) [inline sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala:595]: I don't get the test title?

**Review by viirya — APPROVED** [review]
Just one question about the test.

File: sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala (hunk 1 above)
Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala:595]: It's for testing this code path: https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala#L342

Comment (viirya) [reply sql/core/src/test/scala/org/apache/spark/sql/ExplainSuite.scala:595]: Oh, I see. Thanks.

**Comment by viirya** [comment]
Thanks! Merging to master.

**Pull request closed without merging**